app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -893,6 +895,39 @@ class HomeActivityTest {
     }
   }
 
+  @Test
+  fun testHomeActivity_promotedStoryHasScalableWidth() {
```

Comment:
Please update the PR description with:
- A demonstration that this test fails with the old fix & passes with the new fix
- A demonstration that this entire test suite passes when run on Espresso

Reply:
> > Thanks @vrajdesai78. Took a full pass. Many of the comments are duplicates, but I wanted to make sure each case I found for repeated issues were covered by comments.
> > From a top-level perspective:
> > 
> > * I think the PR title needs to be updated since this PR is now covering more than just the promoted story list fixes
> > * In the original PR there were changes made to app/src/main/res/layout/recently_played_fragment.xml -- why were these dropped here?
> > * Are there any other files that require similar changes (e.g. for toolbars & nav menus)?
> > * How will developers make sure that they know how to address these sp vs. dp cases long-term? Having a test or check seems important here since it's clear that simply reusing an existing pattern may have different results depending on the view being modified.
> 
> @BenHenning I have updated the title as per your suggestion. recently_played_fragment.xml is dropped from this PR because the toolbar from recently_played_fragment is shifted to recently_played_activity and in recently_played_activity.xml layout_height of toolbar is already wrap_content with minHeight. For tests, I am not able to retrieve the exact width of promoted_story_card inside a test. With my current implementation, everytime I run tests I get 840 width even after chainging it's width from dp to sp.

I think it's possible to programmatically set text scaling, see:
https://github.com/oppia/oppia-android/blob/5cd31c36658a8fb840c0b1fc8209726d34d2529a/app/src/main/java/org/oppia/android/app/utility/FontScaleConfigurationUtil.kt#L16

You can also, in Robolectric, specify the smallest width size and the display density for the "device" being run, so you should be able to set up the environment to fit different sized versions of the app (in case you need a larger window for the sp-based elements to grow correctly). This may only work in Robolectric (not sure since the font scaling should, theoretically, work in Espresso), but that should be fine for this regression test.

Reply:
Thanks @BenHenning for your suggestion, I have used adjustFontScale and updated the test accordingly. Now the test is failing when we set width in dp instead of sp. But as you said it works only on robolectric not in espresso.